util/src/test/java/com/redhat/lightblue/util/metrics/DropwizardRequestMetricsTest.java

```diff
@@ -0,0 +1,115 @@
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.redhat.lightblue.util.Error;
+
+public class DropwizardRequestMetricsTest {
+    // Use fresh registry for each test
+    private MetricRegistry metricsRegistry = new MetricRegistry();
+    private RequestMetrics requestMetrics = new DropwizardRequestMetrics(metricsRegistry);
+
+    @Test
+    public void testStartRequestMonitoring() {
+        requestMetrics.startEntityRequest("bulk", null, null);
+        Assert.assertNotNull(metricsRegistry.getCounters());
+        Assert.assertNotNull(metricsRegistry.getTimers());
+
+        Counter activeRequestCounter = metricsRegistry.counter("api.bulk.requests.active");
+        Timer completedRequestTimer = metricsRegistry.timer("api.bulk.requests.completed");
+
+        Assert.assertEquals(1, activeRequestCounter.getCount());
+        Assert.assertEquals(0, completedRequestTimer.getCount());
+        Assert.assertNotNull(completedRequestTimer.getMeanRate());
+    }
+
+    @Test
+    public void testEndRequestMonitoring() {
+        DropwizardRequestMetrics.Context context = requestMetrics.startEntityRequest("explain", "name", "version");
+        context.endRequestMonitoring();
+
+        Counter activeRequestCounter = metricsRegistry.counter("api.explain.name.version.requests.active");
+        Timer completedRequestTimer = metricsRegistry.timer("api.explain.name.version.requests.latency");
+
+        Assert.assertEquals(0, activeRequestCounter.getCount());
+        Assert.assertEquals(1, completedRequestTimer.getCount());
+        Assert.assertNotNull(completedRequestTimer.getOneMinuteRate());
+    }
+
+    @Test
+    public void testMarkRequestExceptionWithError() {
+        DropwizardRequestMetrics.Context context = requestMetrics.startStreamingEntityRequest("find", "name", "version");
+        context.markRequestException(Error.get("errorCode"));
+        
+        Meter exceptionMeter = metricsRegistry.meter("api.stream.find.name.version.requests.exception.Error.errorCode");
+        Assert.assertEquals(1, exceptionMeter.getCount());
+    }
+
+    @Test
+    public void testMarkRequestExceptionWithException() {
+        DropwizardRequestMetrics.Context context = requestMetrics.startBulkRequest();
+        context.markRequestException(new Exception("bulk request exception"));
+        Meter exceptionMeter = metricsRegistry.meter("api.bulk.requests.exception.Exception.bulk.request.exception");
+        Assert.assertEquals(1, exceptionMeter.getCount());
+    }
+    
+    @Test
+    public void testMarkRequestExceptionWithRegex() {
+        DropwizardRequestMetrics.Context context = requestMetrics.startLockRequest("lockcommand", "domain");
+        context.markRequestException(new NullPointerException("this@tests$the_regex#pattern"));
+        Meter exceptionMeter = metricsRegistry.meter("api.lock.domain.lockcommand.requests.exception.NullPointerException.this.tests.the.regex.pattern");
```

Comment:
Like I mentioned above, it could get really messy to use the entire error message in the metric, so this test is invalid for that reason. However, I would still test the regex handling just with Error's error codes instead.

Reply:
Updated test cases and included some new tests for better coverage.